File: Medium/1442. Count Triplets That Can Form Two Arrays of Equal XOR.java
class Solution {
    public int countTriplets(int[] arr) {
        
        int[] prefixXOR = new int[arr.length + 1];
        prefixXOR[0] = 0;
        System.arraycopy(arr, 0, prefixXOR, 1, arr.length);
        
        int size = prefixXOR.length;

        // perform XOR on consecutive elements in the modified array
        for (int i = 1; i < size; i++) {
            prefixXOR[i] ^= prefixXOR[i - 1];
        }

        int count = 0;

        // iterate througth the modified array to count triplets
        for (int start = 0; start < size; start++) {
            for (int end = start + 1; end < size; end ++) {
                if (prefixXOR[start] == prefixXOR[end]) {
                    count += end - start - 1;
                }
            }
        }

        return count;
    }
}

File: Medium/1442. Count Triplets That Can Form Two Arrays of Equal XOR.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        int count = 0;
        // iterating thorough the possible starting index i
        for (int start = 0; start < arr.size(); ++start) {
            // xorA from the start of the array to the mid - 1
            int xorA = 0;

            // iterate thorough each possible index j
            for (int mid = start + 1; mid < arr.size(); ++mid) {
                // update xorA to include arr[mid - 1]
                xorA ^= arr[mid - 1];

                // create xorB for the subarray from mid to the end ot the array
                int xorB = 0;

                for (int end = mid; end < arr.size(); ++end) {
                    xorB ^= arr[end];

                    // found triplet (start, mid, end)
                    if (xorA == xorB) {
                        ++count;
                    }
                }
            }
        }

        // return number of triplets
        return count;
    }
};